src/main/java/com/programmerdan/minecraft/simpleadminhacks/hacks/GameFeatures.java

```diff
@@ -536,52 +545,106 @@ public void onPlayerDeath(PlayerDeathEvent event) {
 	}
-	
+
+	@EventHandler(priority = EventPriority.HIGH)
+	public void onBlockBreak(BlockBreakEvent event) {
+		if (event.getBlock().getType() == Material.RESPAWN_ANCHOR) {
+			checkForAnchors(event.getBlock(), true);
+		}
+	}
+
 	@EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
-	public void hitGoldBlock(PlayerInteractEvent event) {
+	public void hitAnchorTp(PlayerInteractEvent event) {
+		if (!config.isEnabled() || !config.isAnchorblockTeleport()) {
+			return;
+		}
+
 		if (event.getAction() != Action.LEFT_CLICK_BLOCK) {
 			return;
 		}
 		Block block = event.getClickedBlock();
 		if (block == null) {
 			return;
 		}
-		if (block.getType() != Material.GOLD_BLOCK) {
+		if (block.getType() != Material.RESPAWN_ANCHOR) {
 			return;
 		}
 		if (event.getPlayer().isSneaking()) {
 			return;
 		}
 		//require player to stand on block
-		if (!event.getPlayer().getLocation().getBlock().getLocation().equals(block.getRelative(BlockFace.UP).getLocation())) {
+		if (!event.getPlayer().getLocation().getBlock().getLocation()
+				.equals(block.getRelative(BlockFace.UP).getLocation())) {
 			return;
 		}
-		for(int y = block.getY() + 1; y <= 255; y++) {
+		for (int y = block.getY() + 1; y <= 255; y++) {
 			if (doTeleport(block, event.getPlayer(), y)) {
 				return;
 			}
 		}
-		for(int y = 0; y < block.getY(); y++) {
+		for (int y = 0; y < block.getY(); y++) {
 			if (doTeleport(block, event.getPlayer(), y)) {
 				return;
 			}
 		}
 		event.getPlayer().sendMessage(ChatColor.RED + "No suitable destination was found");
 	}
-	
+
 	private static boolean doTeleport(Block source, Player player, int y) {
 		Block target = source.getWorld().getBlockAt(source.getX(), y, source.getZ());
-		if (target.getType() != Material.GOLD_BLOCK) {
+		if (target.getType() != Material.RESPAWN_ANCHOR) {
 			return false;
 		}
 		if (!TeleportUtil.checkForTeleportSpace(target.getRelative(BlockFace.UP).getLocation())) {
+			checkForAnchors(target, false);
 			return false;
 		}
+		checkForAnchors(target, false);
 		Location adjustedLocation = target.getLocation().clone();
 		adjustedLocation.add(0.5, 1.02, 0.5);
 		adjustedLocation.setYaw(player.getLocation().getYaw());
 		adjustedLocation.setPitch(player.getLocation().getPitch());
 		player.teleport(adjustedLocation);
 		return true;
 	}
+
+	private static void checkForAnchors(Block anchor, boolean skipLevel) {
+		List<Block> blocks = new LinkedList<>();
+		//Here we're scanning each y level for respawn anchors to add them to the list
+		for (int y = 0; y <= 255; y++) {
+			if (skipLevel) {
+				if (anchor.getY() == y) {
+					y++;
+				}
+			}
+			Block anblock = anchor.getWorld().getBlockAt(anchor.getX(), y, anchor.getZ());
+			if (anblock.getType() == Material.RESPAWN_ANCHOR) {
+				blocks.add(anblock);
+			}
+		}
+		//No other anchors? ok back out
+		if (blocks.size() == 1) {
+			setAnchorCharges(blocks.get(0), 0);
+			return;
+		}
+		for (Block block : blocks) {
```

Comment:
Wouldn't it make sense for the anchor to be lit up if it has a valid target, meaning it can be used, rather than it itself being a valid target destination.